File: Paradigmas de linguagem de programação/Lista de Exercicio - PLP/Questão 01/Go/main.go
package main

import "fmt"

type Carro struct {
	Marca  string
	Modelo string
	Ano    int
}

func (c Carro) Detalhes() {
	fmt.Printf("Marca: %s, Modelo: %s, Ano: %d\n", c.Marca, c.Modelo, c.Ano)
}

func main() {
	carro1 := Carro{Marca: "Hyundai", Modelo: "HB20x", Ano: 2015}
	carro2 := Carro{Marca: "Fiat", Modelo: "Punto", Ano: 2012}
	carro3 := Carro{Marca: "Ford", Modelo: "EcoSport", Ano: 2019}

	carro1.Detalhes()
	carro2.Detalhes()
	carro3.Detalhes()
}

File: Paradigmas de linguagem de programação/Lista de Exercicio - PLP/Questão 01/Java/Main.java
class Main {
    static class Carro {
        private String marca;
        private String modelo;
        private int ano;

        public Carro(String marca, String modelo, int ano) {
            this.marca = marca;
            this.modelo = modelo;
            this.ano = ano;
        }

        public void detalhes() {
            System.out.println("Marca: " + marca + ", Modelo: " + modelo + ", Ano: " + ano);
        }
    }

    public static void main(String[] args) {
        Carro carro1 = new Carro("Hyundai", "HB20x", 2015);
        Carro carro2 = new Carro("Fiat", "Punto", 2012);
        Carro carro3 = new Carro("Ford", "EcoSport", 2019);

        carro1.detalhes();
        carro2.detalhes();
        carro3.detalhes();
    }
}